rules/prefer-string-slice.js

```diff
@@ -41,10 +82,48 @@ const create = context => {
 				message: 'Prefer `String#slice()` over `String#substring()`.'
 			};
 
-			const canFix = argumentNodes.length === 0;
+			const firstArgument = argumentNodes[0] ? sourceCode.getText(argumentNodes[0]) : undefined;
+			const secondArgument = argumentNodes[1] ? sourceCode.getText(argumentNodes[1]) : undefined;
+
+			const firstNumber = argumentNodes[0] ? getNumericValue(argumentNodes[0]) : undefined;
+
+			let slice;
+
+			if (argumentNodes.length === 0) {
+				slice = [];
+			} else if (argumentNodes.length === 1) {
```

Comment:
too much in this  branch  just  literal an negative , number, otherwise use  math.max

Reply:
Is it the `isLengthProperty()` that you are critical against @fisker?

I ran these modifications on a large internal project at my company and I ran into `.length` properties being used often enough for me to think it merits a plausible conversion.

Reply:
let's wait for @sindresorhus

Reply:
I'm fine with having special logic for `.length`.